src/shared/views/Mapper/Tile.js

```diff
@@ -150,14 +151,38 @@ export class _Tile extends React.Component<Props> {
         });
     };
 
+    parseUrlIfItContainsCredentials = (rawUrl) => {
+        // check if we have a basic http auth scheme in the url and
+        // send the credentials via headers if so, as react-native's
+        // http clients don't seem to support the url scheme.
+        const urlPattern = /^(?<proto>http|https):\/\/(?<username>.+):(?<password>.+)@(?<url>.+)$/;
+        const match = rawUrl.match(urlPattern);
+        if (match) {
+            const { proto, username, password, url } = match.groups;
+            const uri = `${proto}://${url}`;
+            return {
+                uri,
+                headers: {
+                    Authorization: `Basic ${base64.encode(
+                        `${username}:${password}`,
+                    )}`,
+                },
+            };
+        }
+        return { uri: rawUrl };
```

Comment:
this part is only reached, if the pattern don't match, right?

Reply:
yes, correct! The main reason I write it this way is because the linter enforces this pattern.